### PR TITLE
`kanidm_client` bool/return values

### PR DESCRIPTION
**Pull request by QnnOkabayashi**
Fixes https://github.com/kanidm/kanidm/issues/472

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [ ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

My changes were able to satisfy the type checker but I'm having trouble with some JSON somewhere getting deserialized where it expects a `unit` and finds a `bool`. I need help with finding the cause of this and how to fix it. Thanks!

### CODE REVIEW

File: kanidm_client/src/asynchronous.rs
Comment (Firstyear) [inline kanidm_client/src/asynchronous.rs:99]: As much as I love the cleanups, maybe we should all get better at cleaning them in seperate PR's so it's clearer to review. I'm as much guilty, so I need to improve too :)

File: kanidm_client/src/asynchronous.rs
Comment (Firstyear) [inline kanidm_client/src/asynchronous.rs:668]: It may be worth another ticket to remove OperationResponse as I think it does nothing.

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
Do you mind opening a ticket to remove OperationResponse too? Otherwise looks great, I'm happy to merge it if you want. :)

**Comment by QnnOkabayashi** [comment]
> Do you mind opening a ticket to remove OperationResponse too? Otherwise looks great, I'm happy to merge it if you want. :)

This currently fails a bunch of tests, are we ignoring those?

File: kanidm_client/src/asynchronous.rs
Comment (QnnOkabayashi) [reply kanidm_client/src/asynchronous.rs:668]: Agreed, I can open a ticket once this gets merged to avoid conflicts

**Comment by Firstyear** [comment]
> > Do you mind opening a ticket to remove OperationResponse too? Otherwise looks great, I'm happy to merge it if you want. :)
> 
> This currently fails a bunch of tests, are we ignoring those?

I think these failures are introduced because of this change :) 

```
thread 'test_default_entries_rbac_users' panicked at 'called `Result::unwrap()` on an `Err` value: JsonDecode(reqwest::Error { kind: Decode, source: Error("invalid type: boolean `true`, expected unit", line: 1, column: 4) }, "d9ab7178-5124-416a-8183-81d277f1d9a3")', kanidm_client/tests/default_entries.rs:67:56
```

These probably need to be resolved by you before we merge these. Most likely I think that you may have missed bool -> () in async client for idm_group_add_members as I can't see this in your diff?

**Comment by QnnOkabayashi** [comment]
> > > Do you mind opening a ticket to remove OperationResponse too? Otherwise looks great, I'm happy to merge it if you want. :)
> > 
> > 
> > This currently fails a bunch of tests, are we ignoring those?
> 
> I think these failures are introduced because of this change :)
> 
> ```
> thread 'test_default_entries_rbac_users' panicked at 'called `Result::unwrap()` on an `Err` value: JsonDecode(reqwest::Error { kind: Decode, source: Error("invalid type: boolean `true`, expected unit", line: 1, column: 4) }, "d9ab7178-5124-416a-8183-81d277f1d9a3")', kanidm_client/tests/default_entries.rs:67:56
> ```
> 
> These probably need to be resolved by you before we merge these. Most likely I think that you may have missed bool -> () in async client for idm_group_add_members as I can't see this in your diff?

I did change `idm_group_add_members` in `asynchronous.rs` [here](https://github.com/kanidm/kanidm/pull/479/files/3c5cc00a3e55db1b04a50b8acc7e1779eeca797a#diff-65b0009dfc1559162e33b7c9e43e1d45c2857c9c181423210c86dee3770fe368R721-R724), the function name is just not previews so it doesn't show when doing cmd+F. 

The error msg makes me thing there's a JSON object that's being loaded, which makes sense because it makes it past the type checker as a runtime error. The issue is I don't even know where this JSON object is coming from.

**Pull request merged**
